go.mod

```diff
@@ -17,7 +19,7 @@ require (
 	go.opentelemetry.io/otel v1.19.0
 	golang.org/x/exp v0.0.0-20231006140011-7918f672742d
 	k8s.io/api v0.26.2
-	k8s.io/apimachinery v0.26.2
+	k8s.io/apimachinery v0.28.2
 	k8s.io/apiserver v0.26.2
 	k8s.io/client-go v0.26.2
```

Comment:
question: why is this bump needed?

Bumping `apimachinery` usually requires bumping `client-go`, too. Why do we need to use `apimachinery v0.28.2` version for this PR? And why don’t we bump the `go-client` with it, too?

JFYI, we might need to test this change with components that import the Storage Client, because what often happens when bumping `apimachinery`, especially without `k8s.io/client-go` is that clients start sending malformed requests and misbehave. So if we bump `k8s.io/apimachinery`, we need to bump `k8s.io/client-go`. And we also need to bump the Storage client on every single client that imports the Storage client anyway.

But then we will need to manually test the clients that import the Storage client, because `apimachinery` bumps can be breaking.

Reply:
Same reason as above, hadn't made this change explicitly, it got pulled when I moved from 1.20 to 1.21

Reply:
I see. If I were you, I would double-check that `go mod tidy` always bumps `apimachinery`. If it does, then I would bump `client-go` to `v0.28.2`, and try it out with the bumped modules. If all works, then yay. If not, then we will have to pin the apimachinery code.